addons/pkg-maven/ftests/src/main/java/org/commonjava/indy/pkg/maven/content/GroupMetadataMergeWithHangRemoteTest.java

```diff
@@ -0,0 +1,152 @@
+                    + "    <release>%version%</release>\n" + "    <versions>\n" + "      <version>%version%</version>\n"
+                    + "    </versions>\n" + "    <lastUpdated>20150722164334</lastUpdated>\n" + "  </versioning>\n"
+                    + "</metadata>\n";
+    /* @formatter:on */
+
+    private static final String GROUP_METADATA_CONTENT =
+            "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n" + "<metadata>\n" + "  <groupId>org.foo</groupId>\n"
+                    + "  <artifactId>bar</artifactId>\n" + "  <versioning>\n" + "    <latest>1.0</latest>\n"
+                    + "    <release>1.0</release>\n" + "    <versions>\n" + "      <version>1.0</version>\n"
+                    + "    </versions>\n" + "    <lastUpdated>20150722164334</lastUpdated>\n" + "  </versioning>\n"
+                    + "</metadata>\n";
+    /* @formatter:on */
+
+    @Test
+    public void remoteWithSlowResponse()
+            throws Exception
+    {
+        String message = "test metadata merge";
+
+        RemoteRepository remote = new RemoteRepository( REMOTE_A1_NAME, server.formatUrl( REMOTE_A1_NAME ) );
+        remote.setTimeoutSeconds( 600 );
+        RemoteRepository a = client.stores().create( remote, message, RemoteRepository.class );
+        server.expect( "GET", server.formatUrl( REMOTE_A1_NAME, METADATA_PATH ), 408, "Request Time-out" );
+
+        HostedRepository b =
+                client.stores().create( new HostedRepository( HOSTED_B_NAME ), message, HostedRepository.class );
+        deployContent( b, METADATA_PATH, REPO_METADATA_TEMPLATE, B_VERSION );
+
+        Group g = client.stores().create( new Group( GROUP_G1_NAME, a.getKey(), b.getKey() ), message, Group.class );
+
+        assertContent( g, METADATA_PATH, GROUP_METADATA_CONTENT );
+    }
+
+    @Test
+    public void remoteWithTimeout()
+            throws Exception
+    {
+        String message = "test metadata merge";
+
+        RemoteRepository remote = new RemoteRepository( REMOTE_A2_NAME, server.formatUrl( REMOTE_A2_NAME ) );
+        remote.setTimeoutSeconds( 2 );
+
+        RemoteRepository a = client.stores().create( remote, message, RemoteRepository.class );
+        server.expect( "GET", server.formatUrl( REMOTE_A2_NAME, METADATA_PATH ), 200,
+                       new DelayInputStream( 1000 * 5 ) );
+
+        HostedRepository b =
+                client.stores().create( new HostedRepository( HOSTED_B_NAME ), message, HostedRepository.class );
+        deployContent( b, METADATA_PATH, REPO_METADATA_TEMPLATE, B_VERSION );
+
+        Group g = client.stores().create( new Group( GROUP_G2_NAME, a.getKey(), b.getKey() ), message, Group.class );
+
+        assertContent( g, METADATA_PATH, GROUP_METADATA_CONTENT );
+    }
+
+    //    @Test
+    // This is used to test the hanging remote so need time cost, also need to enlarge the indy client RequestTimeoutSeconds
+    public void remoteWithHang()
```

Comment:
I saw you commented this. Maybe just delete it? I think the second case covers the 'hang'. imo, it is hanging when the timeout < delay.

Reply:
@ruhan1 The second case is delay(5s)>repo timeout(2s), the repo will be disabled. The third case is a little different, delay(3mins) < repo timeout(10mins, same with koji undertow 1.4.11 set), but > merge lock timeout(although it's now 4 mins, I adjusted it to 2 mins during testing, because JUnit has a test timeout of 4 mins limitation, so I tweak the lock time by need manually), since the third one will take a long time to verify, so I comment out.

Reply:
For the first request, it should go running the merge function without lock timeout. The lock timeout should only affects subsequent requests... So I guess this case will wait for 3min 'slow response' and return. Am I right?

Reply:
@ruhan1, yeah, it's not blocked, just returns group meta as expected after 3 mins no matter what the value of lock timeout sets.

Reply:
Cool. So maybe we can rename it, how about 'remoteWithSlowResponse'; and rename the first case to 'remoteWithUpstreamTimeout'. wdyt?

Reply:
@ruhan1, +1, update the names, thanks!